client/components/signup-form/index.jsx

```diff
@@ -476,4 +478,9 @@ class SignupForm extends Component {
 	}
 }
 
-export default localize( SignupForm );
+export default connect(
+	null,
+	dispatch => ( {
+		trackLoginMidFlow: () => dispatch( recordTracksEvent( 'calypso_signup_login_midflow' ) )
```

Comment:
I don't think "mid flow" really means at the middle of the flow. See https://en.wiktionary.org/wiki/midflow
Imo `calypso_signup_goto_login_step` is more generic and could be emitted at anypoint when the use clicks on "Log in" from the masterbar.